tree/ntupleutil/v7/src/RNTupleImporterCLI.cxx

```diff
@@ -0,0 +1,205 @@
+   "  --compression, -c compression\n"
+   "    The compression settings of the target RNTuple, provided as an integer.\n"
+   "    This argument should be provided following ROOT's compression setting scheme (algorithm * 100 + level),\n"
+   "    where `algorithm` uses the following mapping: {1: ZLIB, 2: LZMA, 4: LZMA, 5: ZSTD} \n"
+   "    and `level` is a number from 0 to 9 (inclusive, 0 indicating no compression).\n"
+   "    If not specified, the default zstd (505) compression is used.\n"
+   "  --unzipped-page-size size\n"
+   "    The desired approximate unzipped (in-memory) page size in bytes, provided as an integer.\n"
+   "    If not specified, the default size of 64 * 1024 bytes is used.\n"
+   "  --zipped-cluster-size size\n"
+   "    The desired approximate zipped cluster size in bytes, provided as an integer.\n"
+   "    If not specified, the default size of 50 * 1000 * 1000 bytes is used.\n"
+   "  --max-unzipped-cluster-size size\n"
+   "    The desired maximum unzipped (in-memory) cluster size in bytes, provided as an integer.\n"
+   "    If not specified, the default size of 512 * 1024 * 1024 bytes is used.\n"
+   "  --convert-dots\n"
+   "    Whether to convert dots in branch names (if present) to underscores in field names.\n"
+   "    RNTuple does not allow for dots in field names, so this option will convert them to underscores.\n"
+   "    If not specified, no conversion happens and an error is thrown when branch names with dots are encountered.\n"
+   "  --verbose, -v\n"
+   "    Whether to print schema information and progress.\n"
+   "    If not specified, nothing will be printed except for a brief report about the source TTree and target "
+   "RNTuple.\n";
+} // namespace
+
+ImporterConfig ROOT::Experimental::RNTupleImporterCLI::ParseArgs(const std::vector<std::string> &args)
+{
+   // Print help message and exit if "--help"
+   const auto argsProvided = args.size() >= 2;
+   const auto helpUsed = argsProvided && (args[1] == "--help" || args[1] == "-h");
+
+   if (!argsProvided || helpUsed) {
+      std::cout << usageText;
+      if (!argsProvided)
+         std::cout << " Use --help or -h for usage help.";
+      if (helpUsed)
+         std::cout << argUsageText;
+      std::cout << std::endl;
+
+      return {};
+   }
+
+   enum class EArgState {
+      kNone,
+      kTreeName,
+      kTreePath,
+      kNTuplePath,
+      kNTupleName,
+      kCompression,
+      kUnzippedPageSize,
+      kUnzippedClusterSize,
+      kZippedClusterSize
+   } argState = EArgState::kNone;
+   ImporterConfig config;
+   // The config will overwrite the write options set by `RNTupleImporter`, so we need to set the default zstd
+   // compression here as well.
+   config.fNTupleOpts.SetCompression(config.kDefaultCompressionSettings);
+
+   for (size_t i = 1; i < args.size(); ++i) {
+      const auto &arg = args[i];
+
+      if (argState != EArgState::kNone) {
```

Comment:
Why is the `argState` needed?  Wouldn't it be 'simpler' do just assign `config.f...` instead of assigning `argState`?

Reply:
I suppose it's not strictly necessary but in my opinion it makes the loop a bit cleaner (i.e. no need to manually increment `i` when we encounter an option that takes an argument).